plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemoryUsage.hs

```diff
@@ -254,3 +257,15 @@ instance ExMemoryUsage Data where
             List l     -> CostRose 0 $ l <&> sizeData
             I n        -> memoryUsage n
             B b        -> memoryUsage b
+
+instance ExMemoryUsage BLS12_381.G1.Element where
+    memoryUsage _ = singletonRose . unsafeToSatInt $ BLS12_381.G1.memSizeBytes `div` 8
```

Comment:
Chances are, GHC is going to optimize it properly and pull the whole thing to the top level where it can be shared, but I'd still do it manually just to make sure.

Reply:
> I'd still do it manually just to make sure.

Sorry, I'm not sure what you mean.

Reply:
> Sorry, I'm not sure what you mean.

Make it something like

```haskell
g1ElementCost :: CostRose
g1ElementCost = singletonRose . unsafeToSatInt $ BLS12_381.G1.memSizeBytes `div` 8
{-# NOINLINE g1ElementCost #-}

instance ExMemoryUsage BLS12_381.G1.Element where
    memoryUsage _ = g1ElementCost
```

so that the cost is definitely shared and not recomputed each time a `G1.Element` is costed.

Reply:
`quot` could be used instead of `div` but does not matter much here